proxy/proxy.go

```diff
@@ -0,0 +1,19 @@
+package proxy
+
+import (
+	"os"
+	"os/exec"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+func Exec(command string, args []string) error {
+	agent, err := exec.LookPath("picard")
+	if err != nil {
+		return errors.Wrap(err, "Could not find `picard` executable on $PATH; please ensure that build-agent is installed")
+	}
+
+	err = syscall.Exec(agent, args, os.Environ()) // #nosec
```

Comment:
`syscall.Exec` doesn't work on windows (I believe), but that's fine for now.

It seems the args are wrong here
* command is not used anywhere. I noticed that command is currently a string with spaces. `exec` will not perform shell splitting on these args, so it will need to be `[]string{"step", "halt"}`, which I guess needs to get prepended to the `args`
* I believe you also need to prepend `agent` to `args` as well as pass it as the first argument (https://en.wikipedia.org/wiki/Exec_(system_call)#Arguments)

Reply:
Looks like exec uses argv0 for the first arg:

`func Exec(argv0 string, argv []string, envv []string)`